sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -0,0 +1,194 @@
+  MetricsListResponse as GeneratedMetricsListResponse,
+  TimeSeriesElement as GeneratedTimeSeriesElement
+} from "../generated/metrics/src";
+
+import {
+  MetricDefinitionsListOptionalParams as GeneratedMetricDefinitionsListOptionalParams,
+  MetricDefinitionsListResponse as GeneratedMetricDefinitionsListResponse
+} from "../generated/metricsdefinitions/src";
+
+import { MetricNamespacesListResponse as GeneratedMetricNamespacesListResponse } from "../generated/metricsnamespaces/src";
+
+import { formatPreferHeader } from "./util";
+
+import {
+  QueryLogsBatch,
+  QueryLogsBatchResponse,
+  QueryMetricsOptions,
+  QueryMetricsResponse,
+  GetMetricDefinitionsOptions,
+  GetMetricDefinitionsResponse,
+  GetMetricNamespacesResponse,
+  BatchQuery
+} from "../../src";
+import { Metric, TimeSeriesElement } from "../models/publicMetricsModels";
+
+/**
+ * @internal
+ */
+export function convertRequestForQueryBatch(batch: QueryLogsBatch): GeneratedBatchRequest {
+  let id = 0;
+
+  const requests: LogQueryRequest[] = batch.queries.map((query: BatchQuery) => {
+    const body: QueryBody &
+      Partial<
+        Pick<BatchQuery, "includeQueryStatistics" | "serverTimeoutInSeconds" | "workspace">
+      > = { ...query };
+    delete body["workspace"];
+    delete body["serverTimeoutInSeconds"];
+    delete body["includeQueryStatistics"];
+
+    const logQueryRequest: LogQueryRequest = {
+      id: id.toString(),
+      workspace: query.workspace,
+      headers: formatPreferHeader(query),
+      body
+    };
+
+    ++id;
+
+    return logQueryRequest;
+  });
+
+  return {
+    requests
+  };
+}
+
+/**
+ * @internal
+ */
+export function convertResponseForQueryBatch(
+  generatedResponse: GeneratedQueryBatchResponse
+): QueryLogsBatchResponse {
+  const newResponse: QueryLogsBatchResponse = {
+    results: generatedResponse.responses
+      ?.sort((a, b) => {
+        let left = 0;
+        if (a.id != null) {
+          left = parseInt(a.id, 10);
+        }
+
+        let right = 0;
+        if (b.id != null) {
+          right = parseInt(b.id, 10);
+        }
+
+        return left - right;
+      })
+      ?.map((response: LogQueryResponse) => ({
+        id: response.id,
+        status: response.status,
+        // hoist fields from the sub-object 'body' to this level
+        error: response.body?.error,
+        tables: response.body?.tables
+      }))
+  };
+
+  return newResponse;
+}
+
+/**
+ * @internal
+ */
+export function convertRequestForMetrics(
+  queryMetricsOptions: QueryMetricsOptions | undefined
+): GeneratedMetricsListOptionalParams {
+  const obj: GeneratedMetricsListOptionalParams & QueryMetricsOptions = {
+    ...queryMetricsOptions,
+    orderby: queryMetricsOptions?.orderBy,
+    metric: queryMetricsOptions?.metricNames?.join(","),
+    aggregation: queryMetricsOptions?.aggregations?.join(","),
+    metricnamespace: queryMetricsOptions?.metricNamespace
+  };
+
+  delete obj["orderBy"];
```

Comment:
would it make more sense to use destructuring to avoid needing to do this? 

```ts
const { orderBy, metricNames, ...rest } = queryMetricsOptions;
const obj = {
   ...rest,
   orderby: orderBy,
   metric: metricNames?.join(",")
}
```

and so on?

Reply:
Oh wow, yeah, that'd be great :) I'll move over to that, just didn't think of it!

Reply:
Done!